features/flows/classification/engine/impl/src/main/java/org/opennms/netmgt/flows/classification/internal/DefaultClassificationService.java

```diff
@@ -322,7 +331,13 @@ public void validateRule(final Rule validateMe) {
         return sessionUtils.withTransaction(supplier);
     }
 
-    private void updateRulePositionsAndReloadEngine(final List<Rule> rules) {
+    private <T> T runInTransactionAndThenReload(Supplier<T> supplier) {
```

Comment:
While reading these changes, my initial thought was that `runInTransactionAndThenReload` will be extended by the async wrapper. This is clearly not the case. Therefore the whole renaming is purely cosmetic. Please avoid this kind of clutter.

Reply:
I disagree with your judgement. `runInTransactionAndThenReload` combines two steps that are executed at several places. The "reload" was indicated in method names earlier. Therefore I added it as a suffix to the new method.

Reply:
But why have you moved the reload from these methods in the first place? Just to get rid of the double-reload in `updateGroup`?

Reply:
Yes, reload takes several seconds. Classification would be blocked 2 times as long.